### PR TITLE
SNOW-684267 Add column name to data validation error messages

### PR DESCRIPTION
**Pull request by sfc-gh-asen**
cherry picked https://github.com/snowflakedb/snowflake-ingest-java/commit/0364379d5b2093bcab7edccd55b5a16351740091 as a starting point and fixed the merge conflicts

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java [hunk 1]
```diff
@@ -59,14 +59,16 @@ static ParquetBufferValue parseColumnValueToParquet(
           AbstractRowBuffer.ColumnPhysicalType.valueOf(columnMetadata.getPhysicalType());
       switch (typeName) {
         case BOOLEAN:
-          int intValue = DataValidationUtil.validateAndParseBoolean(value);
+          int intValue =
+              DataValidationUtil.validateAndParseBoolean(columnMetadata.getName(), value);
```
Comment (sfc-gh-tzhang) [inline src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java:63]: is this the display name as well?

Comment (sfc-gh-asen) [reply src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java:63]: Yeah, according to https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java#L111 we set the columnMetadata.getName() as the displayName inside the RowBufferStat

**Review by sfc-gh-tjones — APPROVED** [review]
Nice!

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, thanks!

**Pull request merged**